pydantic/_internal/_typing_extra.py

```diff
@@ -205,11 +208,25 @@ def get_cls_type_hints_lenient(obj: Any, globalns: dict[str, Any] | None = None)
     """
     hints = {}
     for base in reversed(obj.__mro__):
-        ann = base.__dict__.get('__annotations__')
-        localns = dict(vars(base))
-        if ann is not None and ann is not GetSetDescriptorType:
-            for name, value in ann.items():
-                hints[name] = eval_type_lenient(value, globalns, localns)
+        new_hints = eval_cls_annotations_lenient(base, globalns)
+        hints.update(new_hints)
+
+    return hints
+
+
+def eval_cls_annotations_lenient(obj: Any, globalns: dict[str, Any] | None = None) -> dict[str, Any]:
+    """Evaluate annotations of a class, (excluding those from parent classes).
+
+    Unlike `typing.get_type_hints`, this function will not error if a forward reference is not resolvable.
+    """
+    hints = {}
+
+    ann = obj.__dict__.get('__annotations__')
```

Comment:
It seems like `__annotations__` for a `TypedDict` always includes all hints for parent classes too, so I think that the diff around this area

```python
>>> class Foo(TypedDict):
...     x: int
...
>>> class Bar(Foo):
...     y: float
...
>>> Bar.__annotations__
{'y': <class 'float'>, 'x': <class 'int'>}
```

I tested this on 3.8 and 3.12, so I assume that it's true on all versions.

If this is the case, then I think this function `eval_cls_annotations_lenient` is not useful here, maybe the changes to this file can just be reverted?